src/charm.py

```diff
@@ -195,6 +257,17 @@ def _on_collect_unit_status(self, event: CollectStatusEvent):  # noqa C901
 
         event.add_status(ActiveStatus())
 
+    def _relation_not_created(self) -> Optional[str]:
```

Comment:
Tis method is strange, I'm not sure I understand its value, it returns one of the not created relations in the list? Why just one and not the list of uncreated relations? In any case we may want to add a docstring to explain it better.

Reply:
The Docstring is added to this method. It returns the relation id as string not in a list.  If all the relations are created, it returns None.

Reply:
Again, Why not return the list of uncreated relations? This is what the PR in the AUSF does and I believe it's a cleaner approach:
- https://github.com/canonical/sdcore-ausf-k8s-operator/pull/196

In either case, let's take the same approach in every charm

Reply:
All the missing relations are caught.